dali/dfu/dfurun.cpp

```diff
@@ -438,17 +438,101 @@ class CDFUengine: public CInterface, implements IDFUengine
     }
 
 
+    // DropZone check
+    void checkFilePath(RemoteFilename & filename)
+    {
+
+        StringBuffer filePath;
+        filename.getLocalPath(filePath);
+        const char * pfilePath = filePath.str();
+
+        if (filename.queryIP().isLoopBack())
+            throwError1(DFTERR_LocalhostAddressUsed, pfilePath);
+
+    #ifdef _DEBUG
+        LOG(MCdebugInfo, unknownJob, "File path is '%s'", filePath.str());
+    #endif
+
+        const char pathSep = filename.getPathSeparator();
+        const char dotString[]    = {pathSep, '.', pathSep, '\0'};
+        const char dotDotString[] = {pathSep, '.', '.', pathSep, '\0'};
+
+        const char * isDotString = strstr(pfilePath, dotString);
+        const char * isDotDotString = strstr(pfilePath, dotDotString);
+        if ((isDotDotString != nullptr) || (isDotString != nullptr))
+            throwError3(DFTERR_InvalidFilePath, pfilePath, dotDotString, dotString);
+
+        Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+        if (factory)
+        {
+            Owned<IConstEnvironment> env = factory->openEnvironment();
+            if (env)
+            {
+                StringBuffer netaddress;
+                filename.queryIP().getIpText(netaddress);
+
+                Owned<IConstDropZoneInfo> dropZone = env->getDropZoneByAddressPath(netaddress.str(), pfilePath);
+                if (!dropZone)
+                {
+                    if (env->isDropZoneRestrictionEnabled())
+                        throwError1(DFTERR_NoMatchingDropzonePath, pfilePath);
+                    else
+                        LOG(MCdebugInfo, unknownJob, "No matching drop zone path to file path: '%s'", pfilePath);
+                }
+                else
+                {
+                    SCMStringBuffer dropZoneName;
+                    dropZone->getName(dropZoneName);
+                    LOG(MCdebugInfo, unknownJob, "Drop zone path '%s' is %svisible in ECLWatch."
+                            , dropZoneName.str()
+                            , (dropZone->isECLWatchVisible() ? "" : "not ")
+                            );
```

Comment:
minor: Probably the same as before, but this is very verbose, tracing every file checked, particularly verbose in the multi filename cases. I think should remove.